decidim-admin/app/packs/src/decidim/admin/admin_autocomplete.js

```diff
@@ -0,0 +1,117 @@
+import AutoComplete from "src/decidim/autocomplete";
+
+/**
+ * This function can be used to create an autocomplete input automatically
+ * from the following kind of div:
+ *   <div data-autocomplete="{...}"></div>
+ *
+ * The data-autocomplete attribute should contain the following configuration
+ * which is used to generate the AutoComplete options:
+ * - name: assembly_member[user_id],
+ * - options: [],
+ * - placeholder: "Select a participant",
+ * - searchPromptText: "Type at least three characters to search"
+ * - noResultsText: "No results found"
+ * - searchURL: "http://..."
+ * - changeURL: null,
+ * - selected: "",
+ *
+ * @param {HTMLElement} el The element to generate the autocomplete for.
+ * @returns {AutoComplete} An instance of the AutoComplete class.
+ */
+const autoConfigure = (el) => {
+  const config = JSON.parse(el.dataset.autocomplete);
+  const input = document.createElement("input");
+  input.name = config.name;
+  input.type = "hidden";
+
+  const textInput = document.createElement("input");
+  input.name = config.name;
+  textInput.type = "text";
+  textInput.className = "autocomplete-input";
+
+  const selectedValue = document.createElement("span");
+  selectedValue.className = "selected-value";
+  selectedValue.style.display = "none";
+
+  const clearSelection = document.createElement("span");
+  clearSelection.className = "clear-selection";
+  clearSelection.innerHTML = "&times;";
+  clearSelection.style.display = "none";
+
+  if (config.placeholder) {
+    textInput.placeholder = config.placeholder;
+  }
+  if (config.selected) {
+    input.value = config.selected.value;
+    textInput.value = config.selected.label;
+  }
+  el.appendChild(textInput);
+  el.insertBefore(input, textInput);
+
+  const dataSource = (query, callback) => {
+    const params = new URLSearchParams({ term: query });
+    fetch(`${config.searchURL}?${params.toString()}`, {
+      method: "GET",
+      headers: { "Content-Type": "application/json" }
+    }).then((response) => response.json()).then((data) => {
+      callback(data)
+    });
+  };
+
+  const ac = new AutoComplete(textInput, {
+    dataMatchKeys: ["label"],
+    dataSource
+  });
+
+  const acWrapper = document.querySelector(".autoComplete_wrapper");
+  acWrapper.insertBefore(clearSelection, textInput);
+  acWrapper.insertBefore(selectedValue, textInput);
+
+  const clearSelected = () => {
+    input.value = ""
+    textInput.placeholder = config.placeholder;
+    clearSelection.style.display = "none";
+    selectedValue.style.display = "none";
+  }
+
+  clearSelection.addEventListener("click", () => {
+    clearSelected();
+  })
+
+  textInput.addEventListener("selection", (event) => {
```

Comment:
I think the single value selector would be very useful to generalize to the main `AutoComplete` class. Although I know the geocoding field works a bit differently as we want to make it possible for the user to edit the address text after they have selected a value from it. But this can be made a configuration option.

We can add a configuration option whether the `AutoComplete` allows defining a `mode` for it:
- `sticky` (default) - Allows selecting a single value and not editing the value after selected (e.g. as the admin autocomplete fields)
- `single` - Allows selecting a single value and editing the selected text after the selection (e.g. geocoding field)
- `multi` - Allows selecting multiple values (e.g. for the `decidim-tags` module, see below)

Multiple selections can otherwise stay as-is because it seems all of the core cases need slightly different rendering / front-end handling. Although, for the multiple value selector, it would be really useful to have a default behavior as well, e.g. for the `decidim-tags` module. It could define the `modifyResult` callback on the fly when (a) the mode is set to `multi` and (b) the `modifyResult` option overridden with the provided configuration options (as e.g. in the participant selector autocomplete).

Here's a Bootstrap example of how it would behave:
https://codepen.io/Kraisorn_R/pen/Ebjdjd

The selected values need to contain (a) the selected value in a hidden input element and (b) the selected value's visible HTML element (e.g. its label). This would be really useful option e.g. for the `decidim-tags` module which is currently based on Tribute. Here you can see how it is adding the hidden inputs and tags to the element on the page:
https://github.com/mainio/decidim-module-tags/blob/main/app/assets/javascripts/decidim/tags/tagging_input.js.es6